TripletCriterion.cu

```diff
@@ -7,21 +7,35 @@
 __global__ void triplet_dist_kernel(const int n, const float norm,
                                     const int nb_batch, const int length,
                                     float* x, float* y) {
+
+  // Unnecessary for loop...
+  // This is EXACTLY equivalent to `if (i < n)` statement
   for (int i = blockIdx.x*blockDim.x+threadIdx.x; i < n; i += blockDim.x*gridDim.x) {
```

Comment:
Yes, it is hard to have the increment of `i` but good to consider the edge case. Nothing is related to shared memory here. Could you remove the comment? unnecessary for loop is unnecessary.